focus-ios/focus-ios-tests/XCUITest/BrowsingTest.swift

```diff
@@ -75,4 +75,67 @@ class BasicBrowsing: BaseTestCase {
         app.buttons["BrowserToolset.stopReloadButton"].tap()
         waitForValueContains(app.textFields["URLBar.urlText"], value: "mozilla")
     }
+
+    // Smoketest
+    // https://testrail.stage.mozaws.net/index.php?/cases/view/2587661
+    func testActivityMenuRequestDesktopItem() {
+        let urlBarTextField = app.textFields["URLBar.urlText"]
+
+        // Wait for existence rather than hittable because the textfield is technically disabled
+        loadWebPage("facebook.com")
+
+        waitForWebPageLoad()
+        waitForExistence(app.buttons["HomeView.settingsButton"])
+        app.buttons["HomeView.settingsButton"].tap()
+
+        if iPad() {
+            waitForExistence(app.collectionViews.buttons["Request Mobile Site"])
+            app.collectionViews.buttons["Request Mobile Site"].tap()
+        } else {
+            waitForExistence(app.collectionViews.buttons["Request Desktop Site"])
+            app.collectionViews.buttons["Request Desktop Site"].tap()
+        }
+
+        waitForWebPageLoad()
+
+        // https://github.com/mozilla-mobile/focus-ios/issues/2782
```

Comment:
@isabelrios Could I remove the code to test on mobile URL? This issue was closed as not fixed. In addition, Firefox iOS could request desktop site but still gets mobile version instead.